x-pack/plugins/security_solution/cypress/integration/exceptions/rule_details_flow/add_exception.spec.ts

```diff
@@ -0,0 +1,221 @@
+import { deleteAlertsAndRules } from '../../../tasks/common';
+import {
+  NO_EXCEPTIONS_EXIST_PROMPT,
+  EXCEPTION_ITEM_VIEWER_CONTAINER,
+  NO_EXCEPTIONS_SEARCH_RESULTS_PROMPT,
+} from '../../../screens/exceptions';
+import {
+  createExceptionList,
+  createExceptionListItem,
+  deleteExceptionList,
+} from '../../../tasks/api_calls/exceptions';
+import { waitForAlertsToPopulate } from '../../../tasks/create_new_rule';
+
+describe('Add exception from rule details', () => {
+  const NUMBER_OF_AUDITBEAT_EXCEPTIONS_ALERTS = '1 alert';
+
+  before(() => {
+    esArchiverResetKibana();
+    esArchiverLoad('exceptions');
+    login();
+  });
+
+  after(() => {
+    esArchiverUnload('exceptions');
+  });
+
+  describe('rule with existing exceptions', () => {
+    const exceptionList = getExceptionList();
+    beforeEach(() => {
+      deleteAlertsAndRules();
+      deleteExceptionList(exceptionList.list_id, exceptionList.namespace_type);
+      // create rule with exceptions
+      createExceptionList(exceptionList, exceptionList.list_id).then((response) => {
+        createCustomRule(
+          {
+            ...getNewRule(),
+            customQuery: 'agent.name:*',
+            dataSource: { index: ['exceptions*'], type: 'indexPatterns' },
+            exceptionLists: [
+              {
+                id: response.body.id,
+                list_id: exceptionList.list_id,
+                type: exceptionList.type,
+                namespace_type: exceptionList.namespace_type,
+              },
+            ],
+          },
+          '2'
+        );
+        createExceptionListItem(exceptionList.list_id, {
+          list_id: exceptionList.list_id,
+          item_id: 'simple_list_item',
+          tags: [],
+          type: 'simple',
+          description: 'Test exception item',
+          name: 'Sample Exception List Item',
+          namespace_type: 'single',
+          entries: [
+            {
+              field: 'user.name',
+              operator: 'included',
+              type: 'match_any',
+              value: ['bar'],
+            },
+          ],
+        });
+        createExceptionListItem(exceptionList.list_id, {
+          list_id: exceptionList.list_id,
+          item_id: 'simple_list_item_2',
+          tags: [],
+          type: 'simple',
+          description: 'Test exception item 2',
+          name: 'Sample Exception List Item 2',
+          namespace_type: 'single',
+          entries: [
+            {
+              field: 'unique_value.test',
+              operator: 'included',
+              type: 'match_any',
+              value: ['foo'],
+            },
+          ],
+        });
+      });
+
+      visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
+      goToRuleDetails();
+      goToExceptionsTab();
+    });
+
+    it('Creates an exception item', () => {
+      // displays existing exception items
+      cy.get(NO_EXCEPTIONS_EXIST_PROMPT).should('not.exist');
```

Comment:
In order to make the test more robust, I would add an assertion to check for an element we know it should be displayed once the page is properly loaded. This is because `goToExceptionsTab();` just clicks on an element, and after that, we are checking for an element to don't exist, it should happen that this assertion is done meanwhile the page is loading, hence, the assertion is going to pass without adding the value we are looking for.

Another thing I believe we can do, is to first run the assertion on line 125 and then this one.

Reply:
Thank you thank you for your feedback. I've opened a follow up PR addressing it [here](https://github.com/elastic/kibana/pull/140448).